src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3836,15 +3837,13 @@ void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTriggerEn
 
         if (scriptholder.event.event_flags & SMART_EVENT_FLAG_DIFFICULTY_ALL)//if has instance flag add only if in it
         {
-            if (obj && obj->GetMap()->IsDungeon())
-            {
-                if ((1 << (obj->GetMap()->GetSpawnMode()+1)) & scriptholder.event.event_flags)
-                {
-                    mEvents.push_back(scriptholder);
-                }
-            }
-            continue;
+            if (!(obj && obj->GetMap()->IsDungeon()))
+                continue;
+
+            if (!(1 << (obj->GetMap()->GetSpawnMode() + 1) & scriptholder.event.event_flags))
+                continue;
         }
+        mAllEventFlags |= scriptholder.event.event_flags;
```

Comment:
I wonder if we should reset mAllEventFlags before this loop in case SAI table is reloaded at runtime

Reply:
We don't update instance of SmartScript, instead we create new instances
https://github.com/TrinityCore/TrinityCore/blob/bf0975f658d5777f0a2812507ab0cca462b18ea7/src/server/game/AI/SmartScripts/SmartScriptMgr.cpp#L335-L343
I think the creatures still have the old version of SAI after reload